spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java

```diff
@@ -311,21 +312,25 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
                 LOGGER.info("Auto proxy of [{}]", beanName);
             }
             DataSourceProxy dataSourceProxy = DataSourceProxyHolder.get().putDataSource((DataSource) bean);
-            return Enhancer.create(bean.getClass(), (org.springframework.cglib.proxy.MethodInterceptor) (o, method, args, methodProxy) -> {
-                Method m = BeanUtils.findDeclaredMethod(DataSourceProxy.class, method.getName(), method.getParameterTypes());
-                if (null != m) {
-                    return m.invoke(dataSourceProxy, args);
-                } else {
-                    boolean oldAccessible = method.isAccessible();
-                    try {
-                        method.setAccessible(true);
-                        return method.invoke(bean, args);
-                    } finally {
-                        //recover the original accessible for security reason
-                        method.setAccessible(oldAccessible);
+            return Proxy.newProxyInstance(Thread.currentThread().getContextClassLoader(), bean.getClass().getInterfaces(), new InvocationHandler() {
```

Comment:
If the user configures DataSourceProxy and auto-proxy is enabled, DataSourceProxy's DataSource will be proxied.
It is recommended to check beforehand, prompt the user and terminate.

Reply:
fixed